### PR TITLE
Admin statistics/ advanced stats

### PR DESCRIPTION
**Pull request by deepsidhu85**
## Description of changes
What did you change in this pull request?  Provide a description of files changed, user interactions changed, etc.  Include how to test your changes.

Added advanced statistics which display charts for analyses ran, projects created, samples created, users created.

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

~* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~
~* [ ] Tests added (or description of how to test) for any new features.~
~* [ ] User documentation updated for UI or technical changes.~

### CODE REVIEW

File: src/main/java/ca/corefacility/bioinformatics/irida/model/enums/StatisticTimePeriod.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package ca.corefacility.bioinformatics.irida.model.enums;
+
+public enum StatisticTimePeriod {
```
Comment (joshsadam) [inline src/main/java/ca/corefacility/bioinformatics/irida/model/enums/StatisticTimePeriod.java:3]: Comments

Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/model/enums/StatisticTimePeriod.java:9]: Added in 06bf685f3158a23f35de733bead1341113532cd5

**Review by joshsadam — CHANGES_REQUESTED** [review]
Overall I would say good.  We have such different ways of using React context.

File: src/main/java/ca/corefacility/bioinformatics/irida/repositories/analysis/submission/AnalysisSubmissionRepository.java [hunk 2]
```diff
@@ -136,28 +125,72 @@
...
 	@Query("select count(s.id) from AnalysisSubmission s where s.createdDate >= ?1")
 	public Long countAnalysesRanInTimePeriod(Date createdDate);
+
+	/**
+	 * Get a list of {@link GenericStatModel}s for analyses run in the last day and grouped by hour
+	 *
+	 * @param createdDate The minimum created date for the analysis submission
+	 * @return A list of {@link GenericStatModel}s
+	 */
+	@Query("select new ca.corefacility.bioinformatics.irida.ria.web.admin.dto.statistics.GenericStatModel(function('date_format', s.createdDate, '%H:00'), count(s.id))"
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/repositories/analysis/submission/AnalysisSubmissionRepository.java:160]: Using @Query to project the results directly into a dto requires the use of the fully classified class name

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/admin/dto/statistics/SampleStatsResponse.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package ca.corefacility.bioinformatics.irida.ria.web.admin.dto.statistics;
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/admin/dto/statistics/SampleStatsResponse.java:1]: I will take a look and see what I can do

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/admin/AdminStatisticsAjaxController.java [hunk 4]
```diff
@@ -44,18 +30,8 @@ public AdminStatisticsAjaxController(ProjectService projectService, UserService
...
-				.toDate();
-
-		Long analysesRan = analysisSubmissionService.getAnalysesRan(minimumCreatedDate);
-		Long projectsCreated = projectService.getProjectsCreated(minimumCreatedDate);
-		Long samplesCreated = sampleService.getSamplesCreated(minimumCreatedDate);
-		Long usersCreated = userService.getUsersCreatedInTimePeriod(minimumCreatedDate);
-		Long usersLoggedIn = userService.getUsersLoggedIn(minimumCreatedDate);
-
-		return ResponseEntity.ok(new BasicStats(analysesRan, projectsCreated, samplesCreated, usersCreated, usersLoggedIn));
+	public ResponseEntity<BasicStats> getAdminStatistics(@RequestParam int timePeriod) {
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/admin/AdminStatisticsAjaxController.java:33]: Sure!

[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/admin/AdminStatisticsAjaxController.java]: As per our discussion we are going to leave this

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/admin/dto/statistics/SampleStatsResponse.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/admin/dto/statistics/SampleStatsResponse.java]: Updated in 0d7b38cc8ff91e129bd1a42729850e14f23871ab

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/admin/AdminStatisticsAjaxController.java [hunk 5]
```diff
@@ -22,18 +15,11 @@
...
 
-	private ProjectService projectService;
-	private UserService userService;
-	private SampleService sampleService;
-	private AnalysisSubmissionService analysisSubmissionService;
+	private UIAdminStatisticsService uiAdminStatisticsService;
 
 	@Autowired
-	public AdminStatisticsAjaxController(ProjectService projectService, UserService userService,
-			SampleService sampleService, AnalysisSubmissionService analysisSubmissionService) {
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/admin/AdminStatisticsAjaxController.java:32]: Thanks :)

**Review by joshsadam — APPROVED** [review]
Great job!

**Pull request merged**